packages/components/src/custom-select-control-v2/test/index.tsx

```diff
@@ -0,0 +1,176 @@
+		const user = userEvent.setup();
+
+		const defaultValues = [
+			'incandescent glow',
+			'ultraviolet morning light',
+		];
+
+		render(
+			<CustomSelect defaultValue={ defaultValues } label="Multi-select">
+				{ [
+					'aurora borealis green',
+					'flamingo pink sunrise',
+					'incandescent glow',
+					'rose blush',
+					'ultraviolet morning light',
+				].map( ( item ) => (
+					<CustomSelectItem key={ item } value={ item }>
+						{ item }
+					</CustomSelectItem>
+				) ) }
+			</CustomSelect>
+		);
+
+		const currentSelectedItem = screen.getByRole( 'combobox', {
+			expanded: false,
+		} );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length } items selected`
+		);
+
+		await user.click( currentSelectedItem );
+
+		expect( screen.getByRole( 'listbox' ) ).toHaveAttribute(
+			'aria-multiselectable'
+		);
+
+		defaultValues.map( ( value ) =>
+			expect(
+				screen.getByRole( 'option', { name: value, selected: true } )
+			).toBeVisible()
+		);
+
+		const nextSelection = screen.getByRole( 'option', {
+			name: 'rose blush',
+		} );
+
+		await user.click( nextSelection );
+
+		expect( nextSelection ).toHaveAttribute( 'aria-selected' );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length + 1 } items selected`
+		);
+	} );
+
+	it( 'Should be able to deselect items when provided an array', async () => {
+		const user = userEvent.setup();
+
+		const defaultValues = [
+			'aurora borealis green',
+			'incandescent glow',
+			'key lime green',
+			'rose blush',
+			'ultraviolet morning light',
+		];
+
+		render(
+			<CustomSelect defaultValue={ defaultValues } label="Multi-select">
+				{ defaultValues.map( ( item ) => (
+					<CustomSelectItem key={ item } value={ item }>
+						{ item }
+					</CustomSelectItem>
+				) ) }
+			</CustomSelect>
+		);
+
+		const currentSelectedItem = screen.getByRole( 'combobox', {
+			expanded: false,
+		} );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length } items selected`
+		);
+
+		await user.click( currentSelectedItem );
+
+		expect( screen.getByRole( 'listbox' ) ).toHaveAttribute(
+			'aria-multiselectable'
+		);
+
+		defaultValues.map( ( option ) =>
+			expect(
+				screen.getByRole( 'option', { name: option, selected: true } )
+			).toBeVisible()
+		);
+
+		const nextSelection = [
+			'aurora borealis green',
+			'rose blush',
+			'incandescent glow',
+		];
+
+		for ( let i = 0; i < nextSelection.length; i++ ) {
+			await user.click(
+				screen.getByRole( 'option', { name: nextSelection[ i ] } )
+			);
+
+			expect(
+				screen.getByRole( 'option', {
+					name: nextSelection[ i ],
+					selected: false,
+				} )
+			).toBeVisible();
+		}
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length - nextSelection.length } items selected`
+		);
+	} );
+
+	it( 'Should show rendered content as selected value when using `renderControlledValue`', async () => {
```

Comment:
I _think_ you meant the prop name here?

```suggestion
	it( 'Should show rendered content as selected value when using `renderSelectedValue`', async () => {
```

Reply:
Related https://github.com/WordPress/gutenberg/pull/58583#discussion_r1475964586